util/file.go

```diff
@@ -146,6 +146,11 @@ func CopyFolderContents(source string, destination string) error {
 		if PathContainsHiddenFileOrFolder(src) {
 			continue
 		} else if file.IsDir() {
+			// Permission and stale cache problems otherwise; see
+			// https://github.com/gruntwork-io/terragrunt/issues/337
+			if err := os.RemoveAll(dest); err != nil {
+				return errors.WithStackTrace(err)
+			}
```

Comment:
Is this the right fix for the permissions issue you were seeing?

I'm looking at how `CopyFolderContents` is used in [download_source](https://github.com/gruntwork-io/terragrunt/blob/0a61ebb2924ac78077f92a29123186b316126635/cli/download_source.go#L59-L61):

```go
util.CopyFolderContents(terragruntOptions.WorkingDir, terraformSource.WorkingDir)
``` 

Let's say you ran Terragrunt from a folder `foo`. If `foo` contained another folder `bar`, then this fix would kick in, deleting `bar`, and taking care of any possible permissions issues. But if `foo` contained a file named `baz`, that file would not be deleted from the previous run (since `file.IsDir()` would be false), and if had some restrictive permission, it would not be overwritten.

So it seems like you would need an `os.Remove` call in the `else` block too, right?

Reply:
Also, could you add an automated test for these cases?